upgradability/set_code_hash/address_store/lib.rs

```diff
@@ -0,0 +1,68 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
+use ink_lang as ink;
+
+#[ink::contract]
+mod address_store {
+    use ink_prelude::vec::Vec;
+    use ink_env::set_code_hash;
+
+    #[ink(storage)]
+    pub struct AddressStore {
+        addresses: Vec<AccountId>,
+    }
+
+    impl AddressStore {
+        /// Initializes empty address store
+        #[ink(constructor)]
+        pub fn new() -> Self {
+            Self {
+                addresses: Vec::new(),
+            }
+        }
+
+        /// Adds new address to store
+        #[ink(message)]
+        pub fn add_new_address(&mut self, address: AccountId) {
+            self.addresses.push(address);
+        }
+
+        /// Returns stored addresses
+        #[ink(message)]
+        pub fn get_addresses(&self) -> Vec<AccountId> {
+            self.addresses.clone()
+        }
+
+        /// Sets new code hash, updates contract code
+        #[ink(message)]
+        pub fn set_code(&mut self, code_hash: [u8; 32]) {
+            set_code_hash(&code_hash).unwrap_or_else(|err| {
```

Comment:
Slightly more rusty way to do it would be to:

- change return type to Result<(), AddressStoreError>
- define and AddressStoreError enum
- use map_err and return some variant (say Error::SetCodeHash)